src/test/java/org/opensearch/sdk/TestExtensionsRunner.java

```diff
@@ -11,55 +11,124 @@
     }
 
     // test manager method invokes start on transport service
     @Test
-    public void testTransportServiceStarted() throws IOException {
-
-        // retrieve and mock transport service
-        TransportService transportService = Mockito.spy(extensionsRunner.createTransportService(settings));
+    public void testTransportServiceStarted() throws Exception {
 
         // verify mocked object interaction in manager method
         extensionsRunner.startTransportService(transportService);
-        Mockito.verify(transportService, times(1)).start();
+        verify(transportService, times(1)).start();
     }
 
     // test manager method invokes accept incoming requests on transport service
     @Test
-    public void testTransportServiceAcceptedIncomingRequests() throws IOException {
-
-        // retrieve and mock transport service
-        TransportService transportService = Mockito.spy(extensionsRunner.createTransportService(settings));
+    public void testTransportServiceAcceptedIncomingRequests() throws Exception {
 
         // verify mocked object interaction in manager method
         extensionsRunner.startTransportService(transportService);
-        Mockito.verify(transportService, times(1)).acceptIncomingRequests();
+        verify(transportService, times(1)).acceptIncomingRequests();
     }
+
+    @Test
+    public void testRegisterRequestHandler() throws Exception {
+
+        extensionsRunner.startTransportService(transportService);
+        verify(transportService, times(3)).registerRequestHandler(anyString(), anyString(), anyBoolean(), anyBoolean(), any(), any());
+    }
+
+    @Test
+    public void testHandlePluginsRequest() throws Exception {
+        DiscoveryNode sourceNode = new DiscoveryNode(
+            "test_node",
+            new TransportAddress(InetAddress.getByName("localhost"), 9876),
+            emptyMap(),
+            emptySet(),
+            Version.CURRENT
+        );
+        PluginRequest pluginRequest = new PluginRequest(sourceNode, null);
+        PluginResponse response = extensionsRunner.handlePluginsRequest(pluginRequest);
+        assertEquals(response.getName(), "RealExtension");
+
+        // Test if the source node is set after handlePluginRequest() is called during OpenSearch bootstrap
+        assertEquals(extensionsRunner.getOpensearchNode(), sourceNode);
+    }
+
+    @Test
+    public void testClusterStateRequest() throws Exception {
+
+        extensionsRunner.sendClusterStateRequest(transportService);
+
+        verify(transportService, times(1)).sendRequest(any(), anyString(), any(), any());
+    }
+
+    @Test
+    public void testClusterSettingRequest() throws Exception {
+
+        extensionsRunner.sendClusterSettingRequest(transportService);
+
+        verify(transportService, times(1)).sendRequest(any(), anyString(), any(), any());
+    }
+
+    @Test
+    public void testLocalNodeRequest() throws Exception {
```

Comment:
You know exactly what comment I'm going to make here so I won't repeat myself :)